server/services/objectAssetService.js

```diff
@@ -2,84 +2,203 @@
+ * @param {number} ensembleIndex - The ensemble index
+ * @returns {string} The cache key for the ensemble index
+ */
 function makeEnsembleCacheKey(ensembleIndex) {
   return `${ENSEMBLE_CACHE}_${ensembleIndex}`;
 }
-async function getAssetByObjectNumber(objectNumber) {
+
+/** Sets the provided artwork asset into the cache
+ * @param {string} objectNumber - The object number for the artwork
+ * @returns {string} The cache key for the artwork, based on the object number
+ */
+function setArtworkInCache(objectNumber, artworkAsset) {
   const objectCacheKey = makeObjectCacheKey(objectNumber);
-  const objectAsset = memoryCache.get(objectCacheKey);
+  memoryCache.put(objectCacheKey, artworkAsset, oneWeek);
+  return objectCacheKey;
+}
+
+/** Get the requested artwork asset from the artwork cache
+ * @param {string} objectNumber - The object number for the artwork
+ * @returns {} The artwork asset from the cache
+ */
+function getArtworkFromCache(objectNumber) {
+  const objectCacheKey = makeObjectCacheKey(objectNumber);
+  return memoryCache.get(objectCacheKey);
+}
+
+/** Higher-level function for retrieving the NetX assets for an artwork given its object number
+ * but with caching functionality implemented.
+ *
+ * If the assets are available in the cache, then they're pulled from there. Otherwise, a fresh
+ * request is made to NetX to retrieve the assets, after which they're placed into the cache
+ *
+ * @param {string} objectNumber - The object number of the artwork to retrieve assets for
+ * @returns The assets for the artwork, pulled from either the cache or NetX directly
+ */
+async function getAssetByObjectNumber(objectNumber) {
+  const artworkAsset = getArtworkFromCache(objectNumber);
 
   // If we don't have a cached version of this object asset
   // let's fetch it live, store it, then return it
-  if (!objectAsset) {
-    const liveObjectAsset = await damsService.getAssetByObjectNumber(
+  if (!artworkAsset) {
+    const liveObjectAsset = await DAMSService.getFullAssetByObjectNumber(
       objectNumber
     );
-    memoryCache.put(objectCacheKey, liveObjectAsset, oneWeek);
+    setArtworkInCache(objectNumber, liveObjectAsset);
 
     return liveObjectAsset;
   }
 
   // Otherwise, we have the cached version
-  return objectAsset;
+  return artworkAsset;
+}
+
+/** Higher-level function for retrieving the NetX assets for multiple artworks given a list of artwork objects
+ * but with caching functionality implemented.
+ *
+ * If the assets are available in the cache, then they're pulled from there. Otherwise, a fresh
+ * request is made to NetX to retrieve the assets, after which they're placed into the cache
+ *
+ * @param {Array<{objectId: string, objectNumber: string}>} artworksInformation - The object number of the artwork to retrieve assets for
+ * @returns The assets for the artwork, pulled from either the cache or NetX directly
+ */
+async function getAssetByObjectIds(artworksInformation) {
```

Comment:
Net new function to retrieve artwork assets using Object IDs. They're pulled from cache or NetX as needed, and cached as this happens.